### PR TITLE
CI tests on Julia 1.8, 1.9 + test on Julia nightly every week

### LINKED ISSUE #359: CI test on supported Julia versions?
Perhaps we change this

https://github.com/SpeedyWeather/SpeedyWeather.jl/blob/bdf02345292895570579d87d5de8d832eedbf7e7/.github/workflows/CI.yml#L17

to match

https://github.com/SpeedyWeather/SpeedyWeather.jl/blob/bdf02345292895570579d87d5de8d832eedbf7e7/docs/src/installation.md#L28-L31

### PR DESCRIPTION
**Pull request by milankl**
fixes #359

Is this the correct yaml syntax?

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -14,7 +14,10 @@ jobs:
       - uses: actions/checkout@v2
       - uses: julia-actions/setup-julia@v1
         with:
-          version: 1
+          version:
+            - '1.8'
+            - '1.9'
+            - 'nightly'
```
Comment (navidcy) [inline .github/workflows/CI.yml:20]:
```suggestion
            - 1.8
            - 1.9
            - nightly
```

I think without quotes? ci didn't run before... let's try this way

**Comment by milankl** [comment]
Nope, also not that way 😆

**Comment by milankl** [comment]
Yes, I saw this in some packages, what does matrix.os mean?

**Comment by navidcy** [comment]
I think there is a way to *not* mark the failing on nightly CI as failure. But I don't remember now on top of my head how to do it.

**Comment by white-alistair** [comment]
> I think there is a way to _not_ mark the failing on nightly CI as failure. But I don't remember now on top of my head how to do it.

That sounds like the best of both worlds! Anything to avoid the dreaded :x:

**Comment by navidcy** [comment]
hm... I can't figure it out.
Personally, I'm happy with either; nightly or no nightly!
1.8 and 1.9 was what I was thinking with #359

**Comment by navidcy** [comment]
OK, let's take out nightly!

**Comment by milankl** [comment]
nightly fails because of NetCDF.jl 
```julia
ERROR: LoadError: Failed to precompile NetCDF [30363a11-5582-574a-97bb-aa9a979735b9] to "/home/runner/.julia/compiled/v1.10/NetCDF/jl_mVW2pK".
...
ERROR: LoadError: InitError: could not load library "/home/runner/.julia/artifacts/0a7f440ba143b238cc3dd0def2956d34d76755f3/lib/libnetcdf.so"
libmbedtls.so.13: cannot open shared object file: No such file or directory
```

**Comment by milankl** [comment]
Apparently Flux separates out the nightly CI to an independent github action, let's try what that does!

**Comment by navidcy** [comment]
That’s nice. We now need to somehow make the nightly CI not make the tests fail. We want to be award of it but without the dreaded ❌….

**Comment by milankl** [comment]
But that's what we have now, no? 👇🏼  nightly fails because of something in NetCDF.jl so we have a ❌ but it's still ✅ overall!

File: .github/workflows/nightly-ci.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    branches:
+      - main
+    tags: '*'
+
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
```
Comment (milankl) [inline .github/workflows/nightly-ci.yml:17]: I've added this, but it doesn't seem to do the job?

Comment (navidcy) [reply .github/workflows/nightly-ci.yml:17]: I'm asking at Slack..

**Comment by navidcy** [comment]
We can do something like

https://github.com/JuliaCI/PkgTemplates.jl/blob/master/.github/workflows/JuliaNightly.yml

which is have the Julia Nightly tests run periodically on main (e.g. every week or so). This way we'll get an idea if something breaks but the Julia nightly won't be part of the CI tests that run for every commit/PR.

**Comment by navidcy** [comment]
How about what I have done with https://github.com/SpeedyWeather/SpeedyWeather.jl/blob/mk/testversions/.github/workflows/julianightly.yml?
I have it run once a week.

**Comment by milankl** [comment]
Awesome! This is great. I'll merge 🎉

**Pull request merged**